docs/4-Using_parameters.md

```diff
@@ -0,0 +1,77 @@
+## Use parameters in your code
+
+Parameter containers are meant to be instantiated just once and be available globally for
+the application. Therefore, implementation has been done as follows:
+
+- The instance of a parameter container is accessed via a class method `get()`;
+- The parameter value is then obtained by chaining with the section name and the
+  parameter name;
+- The first invocation of `get()` will create the container instance, try to read values
+  from file and store it for future access in a private module global (a dictionary with
+  the class id as key);
+- A parameter container should not be instantiated directly by client code (although it
+  is possible to do so, e.g. for testing purposes);
+- If needed, you can set the path for the parameter file before the first invocation of
+  `get()`.
```

Comment:
All this documentation is pretty good for thinking about edge cases.

We should probably have errors / at the very least printing if someone calls `set_filepath` after an instance was already created. We should probably either raise an error, or reload the configuration, although the last one might result in very weird behavior in an application!

Reply:
Good idea, added a check and the option to do reload